Framework/Algorithms/src/MonitorEfficiencyCorUser.cpp

```diff
@@ -51,7 +46,19 @@ void MonitorEfficiencyCorUser::exec() {
   double val;
   Strings::convert(m_inputWS->run().getProperty("Ei")->value(), val);
   m_Ei = val;
-  Strings::convert(m_inputWS->run().getProperty("monitor_counts")->value(),
+
+  std::string mon_counts_log;
+
+  // get name of the monitor counts sample log from the instrument parameter
+  // file
+  try {
+    mon_counts_log = getValFromInstrumentDef("monitor_counts_log");
+  } catch (Kernel::Exception::InstrumentDefinitionError) {
+    // the default value is monitor_counts
+    mon_counts_log = "monitor_counts";
+  }
+
+  Strings::convert(m_inputWS->run().getProperty(mon_counts_log)->value(),
```

Comment:
This looks awfully complicated. Have you tried using
```c++
m_monitorCounts = m_inputWS->run().getPropertyValueAsType<double>(mon_counts_log);
```
If the above works, you could make similar changes to the other similar cases within this algorithm.